### ISSUE TITLE
Analyzer for TestCaseAttribute should also work for generic TestFixtures

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit.analyzers#53]
The analyzer should also handle the following (from \nunit\src\NUnitFramework\tests\Internal\TypeParameterUsedWithTestMethod.cs). Currently, it gives a warning `The value of the argument at position 0 cannot be assigned to the argument x`.

```csharp
    [Category("Generics")]
    [TestFixture(typeof(double))]
    public class TypeParameterUsedWithTestMethod<T>
    {
        [TestCase(5)]
        [TestCase(1.23)]
        public void TestMyArgType(T x)
        {
            Assert.That(x, Is.TypeOf(typeof(T)));
        }
    }
```

**Comment by manfred-brands** [comment]
@mikkelbu The current version (4.6.0) of the analyzer doesn't complain about the above.
Type parameter checking was added in #638 for #632.

The analyzer only checks for `class` or `value` constraints not for the actual value of `T`.

Extending your example to add another `TestFixture` attribute, with a different type fails the NUnit Framework as it cannot invoke the test: `Error Message: System.ArgumentException : Object of type 'System.Int32' cannot be converted to type 'System.String'`

```csharp
    [Category("Generics")]
    [TestFixture(typeof(string))]
    [TestFixture(typeof(double))]
    public class TypeParameterUsedWithTestMethod<T>
    {
        [TestCase(5)]
        [TestCase(1.23)]
        public void TestMyArgType(T x)
        {
            Assert.That(x, Is.TypeOf(typeof(T)));
        }
    }
```

Should the analyzer check type `TestFixture` attributes and where should the diagnostics be raised? On the `[TestFixture(typeof(string)]` or the `[TestCase(5)]`?

Before doing that, what is the practical use-case where you allow a `TestFixture` type parameter, but have constants for `TestCase`?

**Comment by mikkelbu** [comment]
Closing with the same explanation as in #58

> @manfred-brands As far as I can remember (this is more a guess than actual recollection) when I overtook the Analyzers project, then I ran the existing analyzers against all the NUnit Framework tests and created a lot of issues - like this and #53 - when the analyzers failed on valid tests. I didn't think much about whether the tests made practical sense or not (most of them did).
>
> I'll close this issue and #53 as I don't think we should spend time on this. If somebody has some real use-cases, then we can reconsider.
